src/pages/index.js

```diff
@@ -1,140 +1,250 @@
   "#profile-edit-modal",
   (formData) => {
-    const title = formData[".profile-title-input"];
-    const description = formData[".profile-description-input"];
-    profileTitle.textContent = title;
-    profileDescription.textContent = description;
-    profileEditPopup.close();
+    handleEditProfileSubmit(formData);
   }
 );
-profileEditPopup.setEventListeners();
 
+profileEditModal.setEventListeners();
+
+//Declare API class:
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "2db58c83-3523-4135-8564-2660432887a4",
+    "Content-Type": "application/json",
+  },
+});
+
+//Declare UserInfo Class:
+const userInfo = new UserInfo({
+  titleSelector: ".profile__title",
+  descriptionSelector: ".profile__description",
+  profileImage: ".profile__image",
+});
+
+//Declare PopupWithForm for addCardModal:
 const addCardPopup = new PopupWithForm("#add-card-modal", (formData) => {
-  const name = formData["input-type-title"];
-  const link = formData["input-type-url"];
-  renderCard(name, link);
-  addCardPopup.close();
+  handleAddCardSubmit(formData);
 });
 addCardPopup.setEventListeners();
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const card = new Card(cardData, "#card-template", handleImageClick);
-      cardSection.addItem(card.getCardElement());
-    },
-  },
-  ".cards__list"
+//declare Edit Image Modal
+const profileImageModal = new PopupWithForm(
+  "#profile-image-modal",
+  (formData) => {
+    handleChangeProfilePicture(formData.url);
+  }
 );
+profileImageModal.setEventListeners();
 
-cardSection.renderItems();
+//Declare Section for Initial Cards
+let cardSection;
 
+//RenderCardFunction
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  cardWrap.prepend(card.getCardElement());
+  const card = createCard(cardData);
+  cardSection.addItem(card);
 }
-
+//Create Card Function
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteButton,
+    handleLikeButton
+  );
+  return card.getCardElement();
+}
+//Click Card Function
 function handleImageClick(link, name) {
   popupWithImage.open({ name, link });
 }
 
-// Form Validators
-const profileEditValidator = new FormValidator(config, profileEditForm);
-const addCardValidator = new FormValidator(config, addCardFormElement);
-profileEditValidator.enableValidation();
-addCardValidator.enableValidation();
+function handleEditProfileSubmit(inputValues) {
+  profileEditModal.renderLoading("Saving...");
+  api
+    .updateUserInfo(inputValues)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+      profileEditValidator.resetValidation();
+    })
+    .catch((err) => {
+      console.log(`Unable to process request, ${err}`);
+    })
+    .finally(() => {
+      profileEditModal.close();
```

Comment:
* You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

**Please, correct it everywhere**